File: frontend/src/lib/api/models/chat-message.ts


/**
 * 
 * @export
 * @interface ChatMessage
 */
export interface ChatMessage {
    /**
     * 
     * @type {number}
     * @memberof ChatMessage
     */
    'id'?: number;
    /**
     * 
     * @type {string}
     * @memberof ChatMessage
     */
    'chatId'?: string;
    /**
     * 
     * @type {number}
     * @memberof ChatMessage
     */
    'senderId'?: number;
    /**
     * 
     * @type {number}
     * @memberof ChatMessage
     */
    'recipientId'?: number;
    /**
     * 
     * @type {string}
     * @memberof ChatMessage
     */
    'content'?: string;
    /**
     * 
     * @type {string}
     * @memberof ChatMessage
     */
    'timestamp'?: string;
    /**
     * 
     * @type {boolean}
     * @memberof ChatMessage
     */
    'isDummy'?: boolean;
}

File: backend/src/main/java/com/snoopiechat/backend/model/ChatMessage.java
package com.snoopiechat.backend.model;

import java.io.Serializable;
import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ChatMessage implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String chatId;
    private Long senderId;
    private Long recipientId;
    private String content;

    private LocalDateTime timestamp;
    
    @Builder.Default
    private Boolean isDummy = false;

}
